### PR TITLE
feat(Drawer): Added start and end to position props, updated resizing to work with RTL

### LINKED ISSUE #9596: Drawer - RTL position prop values, example names, resizable drawer
The following need to be fixed for RTL support

- The resizable drawer does not work in RTL.  Splitter gets moved in opposite direction.
- Consider changing `position` prop enumeration values of `left` and `right` to be more meaningful for both right-to-left and left-to-right languages.
- Also update the examples that refer to "left" or "right".

### PR DESCRIPTION
**Pull request by tlabaj**
**What**: Closes #9596

### CODE REVIEW

File: packages/react-core/src/components/Drawer/DrawerPanelContent.tsx [hunk 1]
```diff
@@ -166,24 +195,31 @@ export const DrawerPanelContent: React.FunctionComponent<DrawerPanelContentProps
...
     const mousePos = controlPosition;
     let newSize = 0;
-    if (position === 'right') {
-      newSize = right - mousePos;
-    } else if (position === 'left') {
-      newSize = mousePos - left;
+    if (position === 'end' || position === 'right') {
+      newSize = end - mousePos;
+    } else if (position === 'start' || position === 'left') {
+      newSize = mousePos - start;
```
Comment (thatblindgeye) [inline packages/react-core/src/components/Drawer/DrawerPanelContent.tsx:222]: Looks like we need to take into account the isRTL var when reassigning newSize here:

```suggestion
    if (position === 'end' || position === 'right') {
      newSize = isRTL ? mousePos - end : end - mousePos;
    } else if (position === 'start' || position === 'left') {
      newSize = isRTL ? start - mousePos : mousePos - start;
```

Right now when I click and hold the splitter to resize the drawer with RTL enabled, it just snaps to the smallest size and I'm unable to resize the drawer. The above I was able to resize things in both RTL and LTR as expected.

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
One change below. Unrelated to the linked issue but it seems like some of our examples are redundant? We have a "basic" example which shows the drawer at the end, but also have an "end" example that shows the same behavior, just with the `position` prop explicitly passed in. Same goes for the inline examples.

**Review by mcoker — COMMENTED** [review]
LGTM other than Eric's comment about the splitter/resizing in RTL.

**Review by thatblindgeye — APPROVED** [review]

**Review by wise-king-sullyman — APPROVED** [review]

**Review by mcoker — APPROVED** [review]
🤩👍

**Pull request merged**